[MRG+1] DOC Clean up datasets loaders as part of the reorganization of the dataset section

jeremiedbb opened the pull request:
This pull request aims to standardize the datasets informations, as part of a more general reorganization of the dataset section in user guide, see #11083.

Currently, information about datasets can be found a different places, depending on the datasets, with some redundancies. It can be in loader function or module docstrings, in the ``DESCR`` attribute of the loaded data, or directly on the dataset loading utilities page.

I propose to store all the information in the ``DESCR`` attribute, for all datasets ; Leave in the docstrings the minimal informations. For now, the dataset loading utilities page will display the same information as the ``DESCR`` attribute. Keeping the whole information on this page is still in discussion in #11083.

During this cleaning I found 2 datasets that are currently not included in the dataset loading utilities page : the california housing dataset and the species distributions dataset.
I didn't find any issue or PR about the species distributions dataset.
About, the california housing dataset, there have been a PR that aimed to add it to the doc, see #10586. I'm not sure it's been continued.

edit: Fixes #10555

jeremiedbb committed:
first clean ups on fetch datasets

jeremiedbb committed:
keep cleaning

jeremiedbb committed:
fix include path

jeremiedbb committed:
standardization

jeremiedbb committed:
fixes and docstirng formatting

jnothman commented:
Were there todos on the datasets/index.rst page unrelated to moving things
into sklearn/datasets/descr, etc? I'd be keen on making the docs *look*
right before we deal with this maintenance issue.​

Perhaps you can cherry-pick relevant commits from #10586 which appears to
be stalled.

jeremiedbb committed:
fix flake8

jeremiedbb committed:
Merge branch 'master' into clean-up-doc-dataset-2

jeremiedbb commented:
Yes the TODOs are mainly add some introductions for the new subsections in index.rst, they are unrelated to moving things into .../descr.
I'll focus on finishing index.rst and leave this one aside in the meanwhile.

qinhanmin2014 commented:
@jeremiedbb Is this ready for review? Or what's the remaining things to do here?

jeremiedbb commented:
@qinhanmin2014 I'm not sure, I have to check cause I haven't been working on that for 2 weeks. I'll ping when I think it's ready. In the meanwhile, feel free to give me your opinion on the general changes I made. I changed a lot of things, moved some files. Do you think it's fine or have I been too hard on cuts ?

jeremiedbb committed:
Merge remote-tracking branch 'upstream/master' into clean-up-doc-dataset-2

jeremiedbb committed:
add dataset characteristics for real world datasets

jeremiedbb commented:
@qinhanmin2014  ready for review :)

aboucaud commented:
PR #11548 adds doc for the california_housing dataset. 
If it is merged before this one, some modifs are still necessary.

aboucaud committed:
add california housing doc

aboucaud committed:
link Bunch DESCR to the doc file content

aboucaud committed:
add more accurate description

aboucaud committed:
remove old descr files

jeremiedbb commented:
@aboucaud so should #11548 be closed ?

aboucaud commented:
yes, done.

jeremiedbb committed:
Merge branch 'master' into clean-up-doc-dataset-2

aboucaud committed:
remove whitespaces

qinhanmin2014 commented on sklearn/datasets/descr/twenty_newsgroups.rst:
```diff
@@ -62,7 +71,7 @@ attribute is the integer index of the category::
   >>> newsgroups_train.target.shape
   (11314,)
   >>> newsgroups_train.target[:10]
-  array([ 7,  4,  4,  1, 14, 16, 13,  3,  2,  4])
```
What's happening here?

jeremiedbb replied:
I have no idea... I corrected it. 
(I may have had errors and thought that the dataset had been modified, but I've just tested it and it works as expected.)

qinhanmin2014 commented on sklearn/datasets/descr/twenty_newsgroups.rst:
```diff
@@ -105,10 +114,10 @@ components by sample in a more than 30000-dimensional space
 (less than .5% non-zero features)::
 
   >>> vectors.nnz / float(vectors.shape[0])
-  159.01327...
```
We generally don't expose so many digits. Try ``# doctest: +ELLIPSIS``

jeremiedbb replied:
did it. But I found that many more results expose all the digits. Should I make the same for all ?

qinhanmin2014 replied:
Yes, please.

qinhanmin2014 reviewed (commented):
This looks pretty good. Thanks a lot @jeremiedbb @aboucaud for your great work and apologies for the delay.
I'll mark it as 0.20 to attract reviewers.

jeremiedbb committed:
comments

jeremiedbb committed:
Merge remote-tracking branch 'origin/clean-up-doc-dataset-2' into clean-up-doc-dataset-2

jeremiedbb committed:
ellipsis

GaelVaroquaux commented:
@qinhanmin2014 : I think that you reviewed this PR and were +1. Could you mark it as :+1:?

GaelVaroquaux commented on sklearn/datasets/descr/covtype.rst:
```diff
@@ -8,10 +8,19 @@ collected for the task of predicting each patch's cover type,
 There are seven covertypes, making this a multiclass classification problem.
 Each sample has 54 features, described on the
-`dataset's homepage <http://archive.ics.uci.edu/ml/datasets/Covertype>`__.
+`dataset's homepage <http://archive.ics.uci.edu/ml/datasets/Covertype>`_.
```
Why did you change this? "__" means anonymous link, and it was probably a good choice here.

jeremiedbb replied:
probably bad copy paste... reverted

jeremiedbb committed:
missed 1 ellipsis

GaelVaroquaux commented on sklearn/datasets/descr/olivetti_faces.rst:
```diff
@@ -1,12 +1,10 @@
-
-`This dataset contains a set of face images`_ taken between April 1992 and April
-1994 at AT&T Laboratories Cambridge. The
+`This dataset contains a set of face images`_ taken between April 1992 and 
```
It's very surprising to me to see a link here.

GaelVaroquaux replied:
Brain fart. Please ignore.

GaelVaroquaux commented on sklearn/datasets/descr/twenty_newsgroups.rst:
```diff
@@ -132,13 +141,11 @@ which is fast to train and achieves a decent F-score::
   >>> pred = clf.predict(vectors_test)
   >>> metrics.f1_score(newsgroups_test.target, pred, average='macro')
-  0.88213...
+  0.88213592402729568
```
I think that the ellipsis was a good idea. We were just lacking the "doctest: +ELLIPSIS" pragma.

GaelVaroquaux commented on sklearn/datasets/descr/twenty_newsgroups.rst:
```diff
@@ -153,10 +160,10 @@ Let's take a look at what the most informative features are:
+  comp.graphics: organization thanks files subject com image lines university edu graphics
+  sci.space: toronto moon gov com alaska access henry nasa edu space
+  talk.religion.misc: article writes kent people christian jesus sandvik edu com god
 
```
Why did this change?

GaelVaroquaux reviewed (changes requested):
A few minor comments.

jeremiedbb committed:
missed changes in twenty...rst

jnothman commented:
That does look strange... especially as predictive performance seems the same...?

jeremiedbb commented:
This as been changed in #11284.
Looking at the diff I haven't been able to find the reason of this change, besides make the tests pass.

It seems that the dataset has been shuffled before #11284 because the target is not the same... But it shouldn't change the feature importances.

jeremiedbb commented:
@GaelVaroquaux I fixed your requested changes. I think it's good to go.
But before can you take a look a the strange behavior I described above ?

jeremiedbb committed:
Merge remote-tracking branch 'upstream/master' into clean-up-doc-dataset-2

qinhanmin2014 commented on sklearn/datasets/descr/twenty_newsgroups.rst:
```diff
@@ -135,10 +144,10 @@ which is fast to train and achieves a decent F-score::
   0.88213...
 
-(The example :ref:`sphx_glr_auto_examples_text_plot_document_classification_20newsgroups.py` shuffles
+(The example :ref:`sphx_glr_auto_examples_text_document_classification_20newsgroups.py` shuffles
```
What's happening here?

qinhanmin2014 commented on sklearn/datasets/descr/twenty_newsgroups.rst:
```diff
@@ -220,4 +230,4 @@ the ``--filter`` option to compare the results.
    * :ref:`sphx_glr_auto_examples_model_selection_grid_search_text_feature_extraction.py`
 
-   * :ref:`sphx_glr_auto_examples_text_plot_document_classification_20newsgroups.py`
+   * :ref:`sphx_glr_auto_examples_text_document_classification_20newsgroups.py`
```
What's happening here?

jeremiedbb replied:
I've had issues with `twenty_newsgroup.rst`. Some changes happened in the file after I started the PR, and I kept using the older version.
I thought I had catch all the changes, but apparently not. It should be ok now.

qinhanmin2014 reviewed (approved):
It's hard to ensure the correctness of every details here but I'll vote +1 here.

jeremiedbb committed:
twenty_newsgroup examples

qinhanmin2014 reviewed (approved):
LGTM, thanks @jeremiedbb @aboucaud for your great work

jnothman commented:
Thanks @jeremiedbb
